### PR TITLE
Write out reports to individual reconstruction derivative folders

### LINKED ISSUE #51: `test_dipy_mapmri` outputs folders with node name and node qsirecon_suffix
## Summary

The test using the `dipy_mapmri` recon specification writes out folders named `qsirecon-DIPYMAPMRI` (based on the qsirecon_suffix) and `qsirecon-mapmri_recon` (based on the name). I think everything should be in `qsirecon-DIPYMAPMRI`.

## Additional details

- QSIRecon version:
- Docker version:
- Singularity version:

### PR DESCRIPTION
**Pull request by tsalo**
Closes #51.

## Changes proposed in this pull request

- Create prep-style `io_spec.json` to control output filenames.
- Replace `ReconDerivativesDataSink` with `DerivativesDataSink` for all non-folder outputs.
- Start adding figures to HTML reports.
- Create `clean_datasinks` function to define the output directory for each reconstruction workflow separately.
- Refactor the AMICO workflow a bit. Sorry, I got carried away with indentation and such.

To do:

- [ ] Add HTML summaries to reports. I need to figure out what info is relevant.

### CODE REVIEW

**Comment by tsalo** [comment]
My PNG-to-SVG conversion is not working very well, unfortunately. I'll probably drop that in favor of opening an issue on Nireports.

File: qsirecon/interfaces/reports.py
Comment (tsalo) [inline qsirecon/interfaces/reports.py:1]: The new interfaces here aren't used yet. I want to tackle that in a separate PR.

**Comment by tsalo** [comment]
I still want to see if I can reorganize the scalar datasink into a short workflow at some point, but the TSV splitter datasink and folder-based recon datasink will be harder.

**Comment by mattcieslak** [comment]
I always prefer workflows too. I ended up with interfaces for these because you can't predict ahead of time what might be produced in the different nodes. But there might be a way! :roadmap:

File: qsirecon/interfaces/reports.py [hunk 1]
```diff
@@ -31,6 +35,45 @@
...
+\t\t\t<li>Denoising Method: {denoise_method}</li>
+\t\t\t<li>Denoising Window: {denoise_window}</li>
+\t\t\t<li>HMC Transform: {hmc_transform}</li>
+\t\t\t<li>HMC Model: {hmc_model}</li>
+\t\t\t<li>DWI series resampled to spaces: T1wACPC</li>
+\t\t\t<li>Confounds collected: {confounds}</li>
+\t\t\t<li>Impute slice threshold: {impute_slice_threshold}</li>
+\t\t</ul>
+{validation_reports}
+"""
```
Comment (mattcieslak) [inline qsirecon/interfaces/reports.py:61]: this is only relevant for preprocessing. We'll have to get creative for how to represent recon workflows here

**Review by mattcieslak — APPROVED** [review]
This is brilliant work. Thank you!

**Pull request merged**